engine/src/QuadTree.cpp

```diff
@@ -63,24 +66,24 @@ bool QuadTree::split() {
                                                        subHeight,
                                                        nodeCapacity, subLevel));
     southWest = std::unique_ptr<QuadTree>(new QuadTree(parentX,
-                             parentY + subHeight,
+                                                       parentY + subHeight,
                                                        subWidth,
                                                        subHeight,
                                                        nodeCapacity, subLevel));
     southEast = std::unique_ptr<QuadTree>(new QuadTree(parentX + subWidth,
-                             parentY + subHeight,
+                                                       parentY + subHeight,
                                                        subWidth,
                                                        subHeight,
                                                        nodeCapacity, subLevel));
 
     isSplit = true;
 
-    std::list<sf::Sprite *> old_nodes = nodes;
+    SpriteList old_nodes = nodes;
```

Comment:
Maybe you can use `list::splice()` because it avoid de-allocating/re-allocating the objects.
http://www.cplusplus.com/reference/list/list/splice/